x-pack/plugins/cases/public/components/case_view/components/custom_fields.tsx

```diff
@@ -97,11 +97,26 @@ const createMissingAndRemoveExtraCustomFields = (
       (customField) => customField.key === confCustomField.key
     );
 
+    let result;
     if (foundCustomField) {
-      return foundCustomField;
+      result = foundCustomField;
+    } else {
+      result = {
+        key: confCustomField.key,
+        type: confCustomField.type,
+        value: null,
+      };
     }
 
-    return { key: confCustomField.key, type: confCustomField.type, value: null };
+    if (
+      result.value === null &&
+      confCustomField.required &&
+      confCustomField?.defaultValue !== undefined
+    ) {
+      result.value = confCustomField.defaultValue;
+    }
+
+    return result as CaseUICustomField;
   });
```

Comment:
I think this can be simplified as

```
const shouldUseDefaultValue = Boolean(confCustomField.required && confCustomField.defaultValue);

    if (foundCustomField) {
      return {
        ...foundCustomField,
        value:
          foundCustomField.value == null && shouldUseDefaultValue
            ? confCustomField.defaultValue
            : foundCustomField.value,
      } as CaseUICustomField;
    }

    return {
      key: confCustomField.key,
      type: confCustomField.type,
      value: shouldUseDefaultValue ? confCustomField.defaultValue : null,
    } as CaseUICustomField;
```

Wdyt?